jkube-kit/resource/helm/src/main/java/org/eclipse/jkube/kit/resource/helm/oci/OCIRegistryAuthenticator.java

```diff
@@ -0,0 +1,154 @@
+/**
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.resource.helm.oci;
+
+import io.fabric8.kubernetes.client.http.HttpClient;
+import io.fabric8.kubernetes.client.http.HttpRequest;
+import io.fabric8.kubernetes.client.http.HttpResponse;
+import io.fabric8.kubernetes.client.utils.Serialization;
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.jkube.kit.common.KitLogger;
+import org.eclipse.jkube.kit.common.util.Base64Util;
+import org.eclipse.jkube.kit.resource.helm.HelmRepository;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.eclipse.jkube.kit.common.util.AsyncUtil.get;
+import static org.eclipse.jkube.kit.common.util.Fabric8HttpUtil.extractAuthenticationChallengeIntoMap;
+import static org.eclipse.jkube.kit.common.util.Fabric8HttpUtil.toFormData;
+
+
+public class OCIRegistryAuthenticator {
```

Comment:
If OCI is authenticated **only** on failure (401), then maybe we want to implement this as an interceptor